packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -64,19 +68,29 @@ export class DetachedFieldIndex {
 			this.revisionTagCodec,
 			this.options,
 		);
-		populateNestedMap(this.detachedNodeToField, clone.detachedNodeToField, true);
+		// populate the rangeMap of detached nodes
+		for (const [major, innerRangeMap] of this.detachedNodeRangeMap) {
+			const clonedRangeMap: RangeMap<ForestRootId> = [];
+			for (const rangeEntry of innerRangeMap) {
+				clonedRangeMap.push({ ...rangeEntry });
+			}
```

Comment:
Could be refactored into a `clone(innerRangeMap)` function that would go into `rangeMap.ts`.
These lines of code may seem too trivial to warrant a helper function, but it tends to make the code more readable, and there actually a very important decision that needs to be encapsulated here: whether to do `clonedRangeMap.push({ ...rangeEntry })` or `clonedRangeMap.push(rangeEntry)` which could then be made faster and terser with `clonedRangeMap = innerRangeMap.slice()`. The former option (currently used in the code here) is obviously slower since it also clones each range entry object, but whether or not the latter option (not cloning each range entry) is safe depends on whether these range entries can ever be mutated. Right now, it looks like they are mutatable (and mutated), so `clonedRangeMap.push({ ...rangeEntry })` is our only option. We probably want to change that in the future, and it will be easier to do if we put this logic in one place in `rangeMap.ts`.